src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -2636,10 +2638,21 @@ private IDisposable TryLinkerHint(XamlObjectDefinition resource, Action<string>
 		private bool ShouldLazyInitializeResource(XamlObjectDefinition resource)
 		{
 			var typeName = resource.Type.Name;
+			var symbol = GetType(resource.Type);
 
-			if (typeName == "Style" || typeName == "ControlTemplate" || typeName == "DataTemplate")
+			if (resource.Members.Any(HasXNameProperty))
 			{
-				// Always lazily initialize styles and templates, since they may be large
+				return false;
```

Comment:
We never lazy init resources that have an `x:Name` ? It's great as it would reduce issues with resource context propagation in `VisualState`, but it could be quite frequent, no?

Reply:
The main difference of using `x:Name` instead of `x:Key` inside a ResourceDictionary is that the former creates a backing field that can be called from code-behind. Since there's no way to know when a field is accessed, it has to be eagerly initialized.

Reply:
We have a unit test for that :)